### PR TITLE
capability to manage branches

### PR DESCRIPTION
**Pull request by mundra-ankur**
- [x] Branch - use branch and if not exists create one
  - [x] Create branch
  - [x]  Use branch
- [x] Commit - commit and push changes
  - [x] Add Files
  - [x] Commit
  - [x] Push

### CODE REVIEW

File: repository/git.go [hunk 1]
```diff
@@ -84,15 +81,61 @@ func (r *Git) Fetch() (err error) {
...
+	cmd.Options.Add("--message", msg)
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+	return r.Push()
+}
+
+// Push changes to server
+func (r *Git) Push() (err error) {
```
Comment (djzager) [inline repository/git.go:125]:
```suggestion
func (r *Git) push() (err error) {
```

File: repository/git.go [hunk 2]
```diff
@@ -116,7 +159,10 @@ func (r *Git) writeConfig() (err error) {
 	if err != nil {
 		return
 	}
-	s := "[credential]\n"
+	s := "[user]\n"
+	s += "name = Konveyor Dev\n"
+	s += "email = konveyor-dev@googlegroups.com\n"
```
Comment (djzager) [inline repository/git.go:164]: Please make sure we have a GH Issue to make this configurable on the credentials directly.

File: repository/subversion.go [hunk 3]
```diff
@@ -75,6 +71,13 @@ func (r *Subversion) Fetch() (err error) {
 	if err != nil {
 		return
 	}
+	return r.Checkout(r.Application.Repository.Branch)
+}
+
+// Checkout Checkouts the repository.
+func (r *Subversion) Checkout(branch string) (err error) {
```
Comment (djzager) [inline repository/subversion.go:78]: Just go ahead and make the methods not on the interface private.

File: repository/subversion.go [hunk 4]
```diff
@@ -84,25 +87,76 @@ func (r *Subversion) Fetch() (err error) {
...
+func (r *Subversion) Branch(name string) error {
+	err := r.Checkout(name)
+	if err != nil {
+		err = r.CreateBranch(name)
+	}
+	return err
+}
+
+// CreateBranch creates a branch with the given name
+func (r *Subversion) CreateBranch(name string) (err error) {
```
Comment (djzager) [inline repository/subversion.go:107]:
```suggestion
func (r *Subversion) createBranch(name string) (err error) {
```

**Review by djzager — COMMENTED** [review]
Looking good.

File: repository/git.go (hunk 2 above)
Comment (mundra-ankur) [reply repository/git.go:164]: https://github.com/konveyor/tackle2-addon/issues/21

**Review by djzager — APPROVED** [review]

**Pull request merged**